### PR TITLE
Force the nextrun calculation to use UTC to be consistent with the DB

### PR DESCRIPTION
**Pull request by discoinfiltrator**
Solves for: https://github.com/humanmade/Cavalcade/issues/74

### CODE REVIEW

**Comment by discoinfiltrator** [comment]
The `phpcs` violations are not from the file that I modified.

**Comment by mackensen** [comment]
This fix resolved the DST scheduling issue in our environment.

**Comment by discoinfiltrator** [comment]
@rmccue can this be looked at so people aren't scrambling when this comes up again in the fall?

**Comment by discoinfiltrator** [comment]
@rmccue @kadamwhite @missjwo what are the next steps to getting this merged? The DST switchover will be happening soon and it would be amazing to prevent this pretty brutal bug from happening to people.

File: inc/class-job.php [hunk 1]
```diff
@@ -81,8 +84,11 @@ public function mark_completed() {
 	}
 
 	public function reschedule() {
-		$this->nextrun = date( MYSQL_DATE_FORMAT, strtotime( $this->nextrun ) + $this->interval );
-		$this->status  = 'waiting';
+		$date = new DateTime( $this->nextrun, new DateTimeZone( 'UTC' ) );
+		$date->add( new DateInterval( 'PT' . $this->interval . 'S' ) );
```
Comment (roborourke) [inline inc/class-job.php:88]: Super minor micro-optimisation for the string concatenation here
```suggestion
		$date->add( new DateInterval( "PT{$this->interval}S" ) );
```

**Review by roborourke — APPROVED** [review]
I think the code here is good. We haven't run into this issue due to setting our server time to UTC but obviously that's not always something that everyone can or should change.

I need to create a tag of the current version first, then we can merge this in and test it out without potentially affecting production instances.

**Comment by discoinfiltrator** [comment]
I also 100% agree that the DB, OS, PHP, etc should be using UTC - but sometimes you inherit large systems that are working with their current setup and it'd be tough to change 🙃. Thanks!

**Pull request merged**